controller-server/src/test/java/com/yahoo/vespa/hosted/controller/certificate/EndpointCertificatesTest.java

```diff
@@ -251,30 +263,31 @@ void includes_application_endpoint_when_declared() {
                                                                            InstanceName.from("main"), 6),
                                             zone2.region().value(), Map.of(InstanceName.from("main"), 2)))
                 .build();
-        ControllerTester tester = new ControllerTester(SystemName.Public);
         EndpointCertificateValidatorImpl endpointCertificateValidator = new EndpointCertificateValidatorImpl(secretStore, clock);
         EndpointCertificates endpointCertificates = new EndpointCertificates(tester.controller(), endpointCertificateMock, endpointCertificateValidator);
-        List<String> expectedSans = List.of(
+        Set<String> expectedSans = new TreeSet<>(List.of(
                 "vlfms2wpoa4nyrka2s5lktucypjtxkqhv.internal.vespa-app.cloud",
                 "a1.t1.g.vespa-app.cloud",
                 "*.a1.t1.g.vespa-app.cloud",
+                "a1.t1.a.vespa-app.cloud",
                 "a1.t1.aws-us-west-2a.r.vespa-app.cloud",
-                "*.a1.t1.aws-us-west-2a.r.vespa-app.cloud",
                 "a1.t1.aws-us-east-1c.r.vespa-app.cloud",
+                "*.a1.t1.a.vespa-app.cloud",
+                "*.a1.t1.aws-us-west-2a.r.vespa-app.cloud",
                 "*.a1.t1.aws-us-east-1c.r.vespa-app.cloud",
                 "a1.t1.aws-us-east-1c.z.vespa-app.cloud",
                 "*.a1.t1.aws-us-east-1c.z.vespa-app.cloud",
                 "a1.t1.us-east-1.test.z.vespa-app.cloud",
                 "*.a1.t1.us-east-1.test.z.vespa-app.cloud",
                 "a1.t1.us-east-3.staging.z.vespa-app.cloud",
                 "*.a1.t1.us-east-3.staging.z.vespa-app.cloud"
-        );
+        ));
         Optional<EndpointCertificateMetadata> endpointCertificateMetadata = endpointCertificates.getMetadata(instance, zone1, applicationPackage.deploymentSpec());
         assertTrue(endpointCertificateMetadata.isPresent());
         assertTrue(endpointCertificateMetadata.get().keyName().matches("vespa.tls.t1.a1.*-key"));
         assertTrue(endpointCertificateMetadata.get().certName().matches("vespa.tls.t1.a1.*-cert"));
         assertEquals(0, endpointCertificateMetadata.get().version());
-        assertEquals(expectedSans, endpointCertificateMetadata.get().requestedDnsSans());
+        assertEquals(expectedSans, new TreeSet<>(endpointCertificateMetadata.get().requestedDnsSans()));
```

Comment:
Does this mean that `requestedDnsSans` can contain duplicates now? Instead of doing this, how about having the `EndpointCertificateMetadata` constructor ensure there are no duplicates since we never want that in the certificate, e.g.
`this.requestedDnsSans = requestedDnsSans.stream().distinct().toList()`
and remove the set wrapping here.

Reply:
No, that was just because I got tired of reading a super-long line, trying to decipher the actual order of arguments that was required >_< I can sort the lists instead ...